src/models/msg/osmosis/gamm/v1beta1/msg_create_pool.ts

```diff
@@ -0,0 +1,31 @@
+import { Categories } from '../../../types';
+
+class MsgCreatePool {
+    public category: Categories;
+    public type: string;
+    public sender: string;
+    public poolParams: any;
+    public poolAssets: boolean;
+    public futurePoolGovernor: string;
+
+    constructor(payload: any) {
+      this.category = 'osmosis';
+      this.type = payload.type;
+      this.sender = payload.sender;
+      this.poolParams = payload.poolParams;
+      this.poolAssets = payload.poolAssets;
+      this.futurePoolGovernor = payload.futurePoolGovernor;
+    }
+
+    static fromJson(json: any) {
+      return new MsgCreatePool({
+        type: json['@type'],
+        sender: json.sender,
+        poolParams: json.poolParams,
+        poolAssets: json.poolAssets,
+        futurePoolGovernor: json.future_pool_governor,
+      });
+    }
+}
+
+export default MsgCreatePool;
```

Comment:
Please read the docs here: https://docs.bigdipper.live/cosmos-based/frontend/customisation

```
All messages must contain the following: category (module), type (message type), json prop and a fromJson static method.
```

I do the json prop instead of `JSON.stringify(<Model>)` because we sometimes change the key names to fit our needs and follow js rules (camelCase) (ex: `futurePoolGovernor: json.future_pool_governor`). If the user wants to see the 'raw' i want to show them `future_pool_governor` and not `futurePoolGovernor`

Same thing with the folder structure method. Let's just split by module

Reply:
Updated! I added json prop to all osmosis messages.